### DISCUSSION TITLE
Support for reading files from a "Requester Pays"  S3 bucket

### DISCUSSION THREAD

**Discussion by AdarshNamala** [duckdb/duckdb#12132] [Ideas]
Support for reading files from "Requester Pays" S3 bucket

AWS doc about the policy: https://docs.aws.amazon.com/AmazonS3/latest/userguide/RequesterPaysBuckets.html

S3 bucket for testing : 
- s3://coiled-datasets-rp/1trc 
- source: https://docs.coiled.io/blog/1trc.html

Accessing the data using Python S3FS and Pandas: 

`
import s3fs
import pandas as pd

_s3fs = s3fs.S3FileSystem(anon=False, requester_pays=True)
file = "s3://coiled-datasets-rp/1trc/measurements-0.parquet"
data_df = pd.read_parquet(file,filesystem=_s3fs)

data_df.shape
`

**Comment by prmoore77** [comment]
I have added a Pull Request to the httpfs extension to support this here: https://github.com/duckdb/duckdb-httpfs/pull/85
